### PR TITLE
Make verification pass when using Image Builder

### PR DESCRIPTION
**Pull request by slaskawi**
Required to make https://github.com/openshift/release/pull/19608 pass﻿

### CODE REVIEW

File: hack/update-clientgen.sh [hunk 1]
```diff
@@ -8,7 +8,7 @@ CODEGEN_PKG=${CODEGEN_PKG:-$(cd ${SCRIPT_ROOT}; ls -d -1 ./vendor/k8s.io/code-ge
 verify="${VERIFY:-}"
 
 for group in ${API_GROUP_VERSIONS}; do
-  bash ${CODEGEN_PKG}/generate-groups.sh "client,lister,informer" \
+  GOFLAGS="" bash ${CODEGEN_PKG}/generate-groups.sh "client,lister,informer" \
```
Comment (slaskawi) [inline hack/update-clientgen.sh:11]: I'm not sure what sort of template you used for generating those, but this should be there. The Image builder uses `GOFLAGS="-mod=vendor", so the client gen complains about its dependencies.

Comment (stlaz) [reply hack/update-clientgen.sh:11]: /hold
this is suspicious

Comment (slaskawi) [reply hack/update-clientgen.sh:11]: The same change is in `update-clientgen.sh` BTW :)

Comment (s-urbaniak) [reply hack/update-clientgen.sh:11]: this script is inspired by https://github.com/openshift/client-go/blob/master/hack/update-codegen.sh. Can we find out why we don't need GOFLAGS="" there?

**Comment by s-urbaniak** [comment]
instead of ignoring vendoring, which we cannot do in openshift, let's investigate the actual failure in https://prow.ci.openshift.org/view/gs/origin-ci-test/pr-logs/pull/openshift_release/19608/rehearse-19608-pull-ci-openshift-cert-manager-operator-master-verify/1409456747843686400:
```
go: inconsistent vendoring in /go/src/github.com/openshift/cert-manager-operator/vendor/k8s.io/code-generator:
	github.com/emicklei/go-restful@v2.9.5+incompatible: is explicitly required in go.mod, but not marked as explicit in vendor/modules.txt
	github.com/go-openapi/spec@v0.19.5: is explicitly required in go.mod, but not marked as explicit in vendor/modules.txt
	github.com/gogo/protobuf@v1.3.2: is explicitly required in go.mod, but not marked as explicit in vendor/modules.txt
	github.com/golang/protobuf@v1.4.3: is explicitly required in go.mod, but not marked as explicit in vendor/modules.txt
	github.com/google/go-cmp@v0.5.4: is explicitly required in go.mod, but not marked as explicit in vendor/modules.txt
	github.com/googleapis/gnostic@v0.4.1: is explicitly required in go.mod, but not marked as explicit in vendor/modules.txt
	github.com/json-iterator/go@v1.1.10: is explicitly required in go.mod, but not marked as explicit in vendor/modules.txt
	github.com/kr/text@v0.2.0: is explicitly required in go.mod, but not marked as explicit in vendor/modules.txt
	github.com/mailru/easyjson@v0.7.0: is explicitly required in go.mod, but not marked as explicit in vendor/modules.txt
	github.com/niemeyer/pretty@v0.0.0-20200227124842-a10e7caefd8e: is explicitly required in go.mod, but not marked as explicit in vendor/modules.txt
	github.com/onsi/ginkgo@v1.11.0: is explicitly required in go.mod, but not marked as explicit in vendor/modules.txt
	github.com/onsi/gomega@v1.7.0: is explicitly required in go.mod, but not marked as explicit in vendor/modules.txt
	github.com/spf13/pflag@v1.0.5: is explicitly required in go.mod, but not marked as explicit in vendor/modules.txt
	github.com/stretchr/testify@v1.6.1: is explicitly required in go.mod, but not marked as explicit in vendor/modules.txt
	golang.org/x/mod@v0.3.1-0.20200828183125-ce943fd02449: is explicitly required in go.mod, but not marked as explicit in vendor/modules.txt
	golang.org/x/net@v0.0.0-20210224082022-3d97a244fca7: is explicitly required in go.mod, but not marked as explicit in vendor/modules.txt
	golang.org/x/sys@v0.0.0-20210426230700-d19ff857e887: is explicitly required in go.mod, but not marked as explicit in vendor/modules.txt
	golang.org/x/text@v0.3.4: is explicitly required in go.mod, but not marked as explicit in vendor/modules.txt
	golang.org/x/tools@v0.1.0: is explicitly required in go.mod, but not marked as explicit in vendor/modules.txt
	google.golang.org/protobuf@v1.25.0: is explicitly required in go.mod, but not marked as explicit in vendor/modules.txt
	gopkg.in/check.v1@v1.0.0-20200227125254-8fa46927fb4f: is explicitly required in go.mod, but not marked as explicit in vendor/modules.txt
	gopkg.in/yaml.v2@v2.4.0: is explicitly required in go.mod, but not marked as explicit in vendor/modules.txt
	k8s.io/gengo@v0.0.0-20201214224949-b6c5ce23f027: is explicitly required in go.mod, but not marked as explicit in vendor/modules.txt
	k8s.io/klog/v2@v2.8.0: is explicitly required in go.mod, but not marked as explicit in vendor/modules.txt
	k8s.io/kube-openapi@v0.0.0-20210305001622-591a79e4bda7: is explicitly required in go.mod, but not marked as explicit in vendor/modules.txt
	sigs.k8s.io/structured-merge-diff/v4@v4.1.0: is explicitly required in go.mod, but not marked as explicit in vendor/modules.txt

	To ignore the vendor directory, use -mod=readonly or -mod=mod.
	To sync the vendor directory, run:
		go mod vendor
make: *** [Makefile:45: verify-scripts] Error 1
make: Target 'verify' not remade because of errors.
{"component":"entrypoint","error":"wrapped process failed: exit status 2","file":"prow/entrypoint/run.go:80","func":"k8s.io/test-infra/prow/entrypoint.Options.Run","level":"error","msg":"Error executing test process","severity":"error","time":"2021-06-28T10:25:09Z"} 
[36mINFO[0m[2021-06-28T10:25:13Z] Ran for 3m21s                                
[31mERRO[0m[2021-06-28T10:25:13Z] Some steps failed:                           
[31mERRO[0m[2021-06-28T10:25:13Z] 
  * could not run steps: step verify failed: test "verify" failed: the pod ci-op-6mmdc239/verify failed after 1m17s (failed containers: test): ContainerFailed one or more containers exited

Container test exited with code 2, reason Error
---
mod, but not marked as explicit in vendor/modules.txt
	golang.org/x/mod@v0.3.1-0.20200828183125-ce943fd02449: is explicitly required in go.mod, but not marked as explicit in vendor/modules.txt
	golang.org/x/net@v0.0.0-20210224082022-3d97a244fca7: is explicitly required in go.mod, but not marked as explicit in vendor/modules.txt
	golang.org/x/sys@v0.0.0-20210426230700-d19ff857e887: is explicitly required in go.mod, but not marked as explicit in vendor/modules.txt
	golang.org/x/text@v0.3.4: is explicitly required in go.mod, but not marked as explicit in vendor/modules.txt
	golang.org/x/tools@v0.1.0: is explicitly required in go.mod, but not marked as explicit in vendor/modules.txt
	google.golang.org/protobuf@v1.25.0: is explicitly required in go.mod, but not marked as explicit in vendor/modules.txt
	gopkg.in/check.v1@v1.0.0-20200227125254-8fa46927fb4f: is explicitly required in go.mod, but not marked as explicit in vendor/modules.txt
	gopkg.in/yaml.v2@v2.4.0: is explicitly required in go.mod, but not marked as explicit in vendor/modules.txt
	k8s.io/gengo@v0.0.0-20201214224949-b6c5ce23f027: is explicitly required in go.mod, but not marked as explicit in vendor/modules.txt
	k8s.io/klog/v2@v2.8.0: is explicitly required in go.mod, but not marked as explicit in vendor/modules.txt
	k8s.io/kube-openapi@v0.0.0-20210305001622-591a79e4bda7: is explicitly required in go.mod, but not marked as explicit in vendor/modules.txt
	sigs.k8s.io/structured-merge-diff/v4@v4.1.0: is explicitly required in go.mod, but not marked as explicit in vendor/modules.txt

	To ignore the vendor directory, use -mod=readonly or -mod=mod.
	To sync the vendor directory, run:
		go mod vendor
make: *** [Makefile:45: verify-scripts] Error 1
make: Target 'verify' not remade because of errors.
```

File: hack/update-clientgen.sh (hunk 1 above)
Comment (slaskawi) [reply hack/update-clientgen.sh]: Fixed with other method.

**Pull request merged**